shell/models/__tests__/management.cattle.io.node.ts

```diff
@@ -0,0 +1,82 @@
+import MgmtNode from '@shell/models/management.cattle.io.node';
+
+describe('class MgmtNode', () => {
+  const t = jest.fn(() => 'baz');
+  const ctx = { rootGetters: { 'i18n/t': t } };
+
+  const resetMocks = () => {
+    // Clear all mock function calls:
+    jest.clearAllMocks();
```

Comment:
Jest has an hook for before/after where you can reset the mocks. This is definitely not the way to do it.

Also may I ask you what is the reason to reset them? It should be sufficient what is defined globally in the Jest setup.

Reply:
You're talking about the beforeEach and afterEach hooks for Jest. I try not to use those unless I have to if for no other reason than because it breaks a linting rule, if using it is a best practice for us then we should probably remove that rule.

This method is pretty much the way described in the eslint-plugin-jest docs concerning this rule and also mirrors another implementation in the code found in `shell/components/nav/WindowManager/__tests__/ContainerShell.test.ts` where it's used for the same purpose.

In my case, I'm resetting the mocks between tests both to keep each test isolated but also because I'm asserting that the `t` getter is only called once when needed. Strictly speaking, I don't need to pass in CTX and/or reset the mock for most of these tests but it was part of my boilerplate for creating new tests.

Reply:
I personally have found that every time I have to place it in a `beforeEach` I am doing something not correct. We should review this if we find more valid cases.

Specifically in your case you have the CTX called multiple times, because it should be instanciated instead of using the same value.

Reply:
I have not reviewed `ContainerShell`, so I cannot talk about it here.